shared/src/http_transport.rs

```diff
@@ -0,0 +1,192 @@
+use ethcontract::jsonrpc as jsonrpc_core;
+use futures::{future::BoxFuture, FutureExt};
+use jsonrpc_core::types::{Call, Output, Request, Value};
+use reqwest::{Client, Url};
+use serde::de::DeserializeOwned;
+use std::{
+    collections::HashMap,
+    sync::{
+        atomic::{AtomicUsize, Ordering},
+        Arc,
+    },
+};
+use web3::{error::Error as Web3Error, helpers, BatchTransport, RequestId, Transport};
+
+#[derive(Clone, Debug)]
+pub struct HttpTransport {
+    client: Client,
+    inner: Arc<Inner>,
+}
+
+#[derive(Debug)]
+struct Inner {
+    url: Url,
+    id: AtomicUsize,
+}
+
+impl HttpTransport {
+    pub fn new(url: Url) -> Self {
+        Self::with_client(Client::new(), url)
+    }
+
+    pub fn with_client(client: Client, url: Url) -> Self {
+        Self {
+            client,
+            inner: Arc::new(Inner {
+                url,
+                id: AtomicUsize::new(0),
+            }),
+        }
+    }
+
+    pub fn next_id(&self) -> RequestId {
+        self.inner.id.fetch_add(1, Ordering::SeqCst)
```

Comment:
Maybe a comment on why this ordering (if I remember we don't care about performance so the most strict one?).

Reply:
This is the most strict ordering (as in always being correct) so I don't think a comment is needed tbh for people that have encountered atomics before.